ideascube/library/models.py

```diff
@@ -99,3 +101,10 @@ def __str__(self):
 
     def get_absolute_url(self):
         return reverse('library:book_detail', kwargs={'pk': self.item.pk})
+
+    @property
+    def extension(self):
+        if not self.file:
+            return ''
+        name, extension = os.path.splitext(self.file.name)
+        return extension[1:]
```

Comment:
Agreed that there are edge cases, but I think whatever the split method we chose we have some.
That said I think we can stay on that simple option, that should cover the expected extension ('.epub', '.pdf'…), and if someone actually upload a complex extension, that will not be such an issue.